### PR TITLE
[plugin.offtictor] 1.0.0

### PR DESCRIPTION
**Pull request by kusinkay**
### Description
An audio plugin to listen to the subscribed feeds registered in your The Old Reader's account.
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.offtictor/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/.project
```
Comment (enen92) [inline plugin.offtictor/.gitignore]: it would be nice to drop this file from the addon PR

File: plugin.offtictor/addon.py [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+    #sCursor = subs.get_unread_cursor()
+    xbmcgui.Dialog().notification(addonname, strings.get('Subscriptions_preloaded'), icon='', time=0, sound=False)
+    feed = sCursor.next()
+    while (feed!=False):
+        if feed.matches:
+            xbmcgui.Dialog().notification(addonname, feed.title, icon='', time=0, sound=False)
+            feed.title += ' (' + str(feed.unread_count) + ')'
+            li = ListItem()
+            li.setLabel(feed.title)
+            li.setIconImage(feed.iconUrl)
```
Comment (enen92) [inline plugin.offtictor/addon.py]: Please use setArt, setIconImage is deprecated

File: plugin.offtictor/addon.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.offtictor" name="The Old Reader addon (by offtic)" version="1.0.0_beta" provider-name="kusinkay">
+	<requires>
+		<import addon="xbmc.python" version="2.14.0"/>
+		<import addon="script.module.addon.common"/>
+		<import addon="script.common.plugin.cache" version="2.5.5"/>
+		<import addon="script.module.pydevd" version="3.4.1"/><!-- debug from linux -->
```
Comment (enen92) [inline plugin.offtictor/addon.xml]: set as optional please

File: plugin.offtictor/addon.xml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.offtictor" name="The Old Reader addon (by offtic)" version="1.0.0_beta" provider-name="kusinkay">
+	<requires>
+		<import addon="xbmc.python" version="2.14.0"/>
```
Comment (enen92) [inline plugin.offtictor/addon.xml]: 2.25.0

File: plugin.offtictor/addon.xml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+		<import addon="script.module.addon.common"/>
+		<import addon="script.common.plugin.cache" version="2.5.5"/>
+		<import addon="script.module.pydevd" version="3.4.1"/><!-- debug from linux -->
+	</requires>
+	<extension point="xbmc.python.pluginsource" library="addon.py">
+		<provides>audio</provides>
+	</extension>
+	<extension point="xbmc.addon.metadata">
+		<platform>all</platform>
+		<summary lang="en">Unofficial The Old Reader RSS add-on</summary>
```
Comment (enen92) [inline plugin.offtictor/addon.xml]: please adopt new language tags:
eg. en_GB

File: plugin.offtictor/resources/language/English/strings.po [hunk 6]
```diff
@@ -0,0 +1,102 @@
+# Kodi Media Center language file
```
Comment (enen92) [inline plugin.offtictor/resources/language/English/strings.po]: resources/language/resource.language.en_gb/strings.po

File: plugin.offtictor/resources/language/Spanish/strings.po [hunk 7]
```diff
@@ -0,0 +1,102 @@
+# Kodi Media Center language file
```
Comment (enen92) [inline plugin.offtictor/resources/language/Spanish/strings.po]: resource.language.es_es/strings.po

File: plugin.offtictor/resources/lib/.gitignore [hunk 8]
```diff
@@ -0,0 +1,6 @@
+/__init__.pyc
```
Comment (enen92) [inline plugin.offtictor/resources/lib/.gitignore]: drop please

File: plugin.offtictor/resources/settings.xml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
```
Comment (enen92) [inline plugin.offtictor/resources/settings.xml]: remove pyc file

File: plugin.offtictor/samples/suscripciones.json [hunk 10]
```diff
@@ -0,0 +1,409 @@
...
+			"title": "9 Sant Vicen\u00e7",
+			"categories": 
+			[
+				
+			],
+
+			"sortid": "51de6561a201dead6f002dbb",
+			"firstitemmsec": "1408531853000",
+			"url": "http://www.blogger.com/feeds/1587207914235716365/posts/default",
+			"htmlUrl": "http://www.nousv.com/",
```
Comment (enen92) [inline plugin.offtictor/samples/suscripciones.json]: doesn't exist

File: plugin.offtictor/resources/lib/tor.py [hunk 11]
```diff
@@ -0,0 +1,419 @@
...
+
+        header = {}
+        if self.auth_code:
+            header['Authorization'] = "GoogleLogin auth=" + self.auth_code
+        header.update(self.header)
+
+        var_json = {'output': 'json'}
+        var_json.update(var)
+        for param in var:
+            self._logger.debug(param + ":" + str(var[param]))
```
Comment (Rechi) [inline plugin.offtictor/resources/lib/tor.py]: don't spam the log

File: plugin.offtictor/addon.py [hunk 12]
```diff
@@ -0,0 +1,338 @@
...
+
+
+# append pydev remote debugger
+if REMOTE_DBG:
+    # Make pydev debugger works for auto reload.
+    # Note pydevd module need to be copied in XBMC\system\python\Lib\pysrc
+    try:
+        #import pysrc.pydevd as pydevd # with the addon script.module.pydevd, only use `import pydevd`
+        import sys
+        sys.path.append('/home/juane/.kodi/addons/script.module.pydevd/lib/')
```
[the remark this answers is not in the document]
Comment (enen92) [reply plugin.offtictor/addon.py]: Ths is only used for debugging, value is always false unless the dev changes (manually) the variable above. It's fine as is for me.

[the remark this answers is not in the document]
Comment (Rechi) [reply plugin.offtictor/addon.py]: - don't add paths which are only valid on your own machine
- this isn't even needed as listing `script.module.pydevd` in addon.xml is the correct way
- is only used for ..., is a bad reason for an exception

File: plugin.offtictor/addon.xml [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+		<description lang="en">It shows your TOR subscriptions on KODI. For each channel you can listen to its podcasts.</description>
+		<summary lang="ca">Connector NO oficial per a RSS de The Old Reader</summary>
+		<description lang="ca">Mostra les teves subscripcions RSS a The Old Reader. Per cada canal pots escoltar els seus podcasts.</description>
+		<summary lang="es">Connector NO oficial para RSS de The Old Reader</summary>
+		<description lang="es">Mostra tus suscripciones RSS en The Old Reader. Para cada canal puedes escuchar sus podcasts.</description>
+		<license>GNU General Public License, v2</license>
+		<language>en ca es</language>
```
[the remark this answers is not in the document]
Comment (kusinkay) [reply plugin.offtictor/addon.xml]: Many thanks for all your clues, @enen92 
This plugin is not about any radio, is for subscribed podcasts and its content language would depend on the The Old Reader's user preferences.

Cheers

File: plugin.offtictor/addon.xml [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+	</requires>
+	<extension point="xbmc.python.pluginsource" library="addon.py">
+		<provides>audio</provides>
+	</extension>
+	<extension point="xbmc.addon.metadata">
+		<platform>all</platform>
+		<summary lang="en">Unofficial The Old Reader RSS add-on</summary>
+		<description lang="en">It shows your TOR subscriptions on KODI. For each channel you can listen to its podcasts.</description>
+		<summary lang="ca">Connector NO oficial per a RSS de The Old Reader</summary>
+		<description lang="ca">Mostra les teves subscripcions RSS a The Old Reader. Per cada canal pots escoltar els seus podcasts.</description>
```
[the remark this answers is not in the document]
Comment (kusinkay) [reply plugin.offtictor/addon.xml]: Your guess is right
;^)

File: plugin.offtictor/samples/suscripciones.json (hunk 10 above)
Comment (kusinkay) [reply plugin.offtictor/samples/suscripciones.json]: samples folder is for a guide to develop.
I'll remove it the next release

File: plugin.offtictor/addon.py [hunk 15]
```diff
@@ -0,0 +1,338 @@
...
+        import pydevd
+        # stdoutToServer and stderrToServer redirect stdout and stderr to eclipse console
+        pydevd.settrace('localhost', stdoutToServer=True, stderrToServer=True)
+    except ImportError:
+        sys.stderr.write("Error: " +
+            "You must add org.python.pydev.debug.pysrc to your PYTHONPATH.")
```
[the remark this answers is not in the document]
Comment (kusinkay) [reply plugin.offtictor/addon.py]: Many thanks @Rechi 
Sorry I don't get it. Shall I make some fixings here?

File: plugin.offtictor/addon.py [hunk 16]
```diff
@@ -0,0 +1,338 @@
...
+    if len(sys.argv)>3:
+        action =sys.argv[3]
+    if len(sys.argv)>4:
+        feed_id =sys.argv[4]
+    
+
+def feed(feedId, next_pointer=None):
+    try:
+        
+        xbmc.log("================= " + addonname + " ========================")
```
[the remark this answers is not in the document]
Comment (kusinkay) [reply plugin.offtictor/addon.py]: Would it be enough to delete all the '='?

[the remark this answers is not in the document]
Comment (Rechi) [reply plugin.offtictor/addon.py]: remove the whole statement, Kodi logs something similar already if an add-on gets invoked

File: plugin.offtictor/addon.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (Rechi) [reply plugin.offtictor/addon.py]: yes replace `sys.stderr.write(...)` with `xbmc.log(...)`

File: plugin.offtictor/addon.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (kusinkay) [reply plugin.offtictor/addon.py]: I agree with @Rechi, I should find another more elegant solution.
Many thanks to both: @enen92 and @Rechi

[the remark this answers is not in the document]
Comment (Rechi) [reply plugin.offtictor/addon.py]: As already mentioned, it should already work without this hack, as you've listed `script.module.pydevd` as dependency in addon.xml.

**Comment by kusinkay** [comment]
Sorry for all the commits.
I'm aware that there should only be one, but I don't know how to follow.
I've read (This info about updating a pull request)[https://kodi.wiki/view/HOW-TO:Update_a_pull_request_to_an_official_addon_repository_on_GitHub] but I'm afraid I'm losing some previous steps.
Appreciate any help.
Many thanks

**Comment by kusinkay** [comment]
I'm so sorry @Rechi  and @enen92 , but I'm not able to update this PR.
Would it be fine to open a new PR?
Many thanks

**Comment by Rechi** [comment]
To update a PR you only have to update the PR branch (krypton) in you repo. Execute the following commands (origin is your repo, upstream is the xbmc repo) to do achieve that.

```
git fetch upstream
git checkout krypton
git reset --hard upstream/krypton
# copy your addon files to plugin.offtictor subdirectory
git add -- plugin.offtictor
git commit -m '[plugin.offtictor] 1.0.0'
git push -f origin krypton
```

**Comment by kusinkay** [comment]
Hi @enen92 and @Rechi 
Are there some more fixing left?
Thank you so much

**Comment by Rechi** [comment]
Still much log spamming left, but not that important.

> Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

You checked that in the PR description, but currently your PR contains 3 commits.

**Comment by kusinkay** [comment]
Oops,
I thought I'd done it the right way this time.
I'm so sorry, @Rechi  It was not easy for me to apply the commands you give me.
I'm closing this PR (I even don't if I have the right to do so) 
Would be a bad idea to open a new PR?
Many thanks

**Pull request closed without merging**